easybuild/easyblocks/p/petsc.py

```diff
@@ -285,16 +288,17 @@ def make_module_req_guess(self):
 
         guesses = super(EB_PETSc, self).make_module_req_guess()
 
-        prefix1 = ''
-        prefix2 = ''
         if self.cfg['sourceinstall']:
-            prefix1 = self.petsc_subdir
-            prefix2 = os.path.join(self.petsc_subdir, self.petsc_arch)
+            self.prefix_inc = self.petsc_subdir
+            self.prefix_lib = os.path.join(self.petsc_subdir, self.petsc_arch)
+
+        if LooseVersion(self.version) >= LooseVersion("3.9"):
+            self.prefix_bin = os.path.join(self.prefix_inc, 'lib', 'petsc')
```

Comment:
@RvDijk This doesn't work currently, because the paths in `sanity_check_step` method are constructed using `self.prefix_*` before `make_module_req_guess` is called.

You should move the code that updates `self.prefix_*` into `__init__` above, right below where you initialize these to `''`.